regulation-worker/internal/delete/batch/batch.go

```diff
@@ -144,115 +113,106 @@ func (*Batch) updateStatusTrackerFile(absStatusTrackerFileName, fileName string)
+		if lines[0] != jobID {
+			f.Close()
 
-// compress & write `cleanedBytes` of type []byte to `fileName`
-func (*Batch) compress(fileName string, cleanedBytes []byte) error {
-	// compressing
-	var buffer bytes.Buffer
-	w := gzip.NewWriter(&buffer)
-	_, err := w.Write(cleanedBytes)
-	if err != nil {
-		return fmt.Errorf("error while compressing file: %w", err)
-	}
-	w.Close() // must close this first to flush the bytes to the buffer.
+			statusTrackerTmpDir, err := os.MkdirTemp(b.TmpDirPath, "")
+			if err != nil {
+				return nil, fmt.Errorf("error while creating temporary directory: %w", err)
+			}
 
-	cleanCompressedFilePtr, err := os.OpenFile(fileName, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o644)
-	if err != nil {
-		return fmt.Errorf("error while opening file, %w", err)
-	}
-	defer cleanCompressedFilePtr.Close()
+			f, err = os.OpenFile(filepath.Join(statusTrackerTmpDir, StatusTrackerFileName), os.O_CREATE|os.O_RDWR, 0o644)
+			if err != nil {
+				return nil, fmt.Errorf("error while opening file to create status tracker, %w", err)
+			}
 
-	_, err = cleanCompressedFilePtr.Write(buffer.Bytes())
-	if err != nil {
-		return fmt.Errorf("error while writing cleaned & compressed data:%w", err)
+			if _, err := io.WriteString(f, jobID+"\n"); err != nil {
+				return nil, fmt.Errorf("error while writing to status tracker file:%s, err: %w", StatusTrackerFileName, err)
+			}
+
+		} else {
+			// if yes, then simply read it.
+			response = lines[1 : len(lines)-1]
+		}
 	}
 
-	return nil
+	return response, nil
 }
 
-// delete users corresponding to `userAttributes` from `fileName` available locally
-func (b *Batch) delete(ctx context.Context, PatternFile, targetFile string) error {
-	decompressedFile, err := b.decompress(targetFile)
+// downloads `fileName` locally. And returns empty file, if file not found.
+// Note: download happens concurrently in 5 go routine by default.
+func (b *Batch) download(_ context.Context, completeFileName string) (string, error) {
+	pkgLogger.Debugf("downloading file: %v", completeFileName)
+
+	tmpFilePathPrefix, err := os.MkdirTemp(b.TmpDirPath, "")
 	if err != nil {
-		return fmt.Errorf("error while decompressing file: %w", err)
+		return "", fmt.Errorf("error while creating temporary directory: %w", err)
 	}
 
-	out, err := b.DM.delete(ctx, PatternFile, decompressedFile)
+	_, fileName := filepath.Split(completeFileName)
+	tmpFilePtr, err := os.OpenFile(filepath.Join(tmpFilePathPrefix, fileName), os.O_CREATE|os.O_RDWR, 0o644)
 	if err != nil {
-		return fmt.Errorf("error while cleaning object, %w", err)
+		return "", fmt.Errorf("error while opening file, %w", err)
 	}
+	defer tmpFilePtr.Close()
 
-	err = b.compress(targetFile, out)
+	absPath, err := filepath.Abs(tmpFilePtr.Name())
 	if err != nil {
-		return fmt.Errorf("error while compressing file: %w", err)
+		return "", fmt.Errorf("error while getting absolute path: %w", err)
 	}
 
-	return nil
+	err = b.FM.Download(context.TODO(), tmpFilePtr, completeFileName)
```

Comment:
We can pass the context here:
```suggestion
	err = b.FM.Download(ctx, tmpFilePtr, completeFileName)
```